src/main/java/seedu/address/model/person/predicates/IncomeComparisonPredicate.java

```diff
@@ -0,0 +1,71 @@
+package seedu.address.model.person.predicates;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.person.Person;
+import seedu.address.model.util.IncomeComparisonOperator;
+
+/**
+ * Predicate that compares a {@code Person}'s income against a threshold using a specified comparison operator.
+ */
+public class IncomeComparisonPredicate implements Predicate<Person> {
+    private final int incomeThreshold;
+    private final IncomeComparisonOperator incomeComparisonOperator;
+
+    /**
+     * Constructs an {@code IncomeComparisonPredicate}.
+     *
+     * @param incomeComparisonOperator The operator used to compare the person's income with the threshold.
+     * @param incomeThreshold          The threshold income to compare against.
+     */
+    public IncomeComparisonPredicate(IncomeComparisonOperator incomeComparisonOperator, int incomeThreshold) {
+        requireNonNull(incomeComparisonOperator);
```

Comment:
Good job on using `requireNonNull` to validate the `incomeComparisonOperator`. Have you considered adding a similar check for `incomeThreshold`? (to ensure it's not negative)

Reply:
Great catch actually, I'll add that in.

Reply:
Great idea, will implement it